### PR TITLE
Add support for DNSoverTLS

### PR DESCRIPTION
**Pull request by shibumi**
systemd-resolved supports DNSOverTLS since version 239.
Current Options for this feature are:

*  'opportunistic'
*  'no'

There will be a 'strict' or 'yes' feature in the future.

Signed-off-by: Christian Rebischke <chris@nullday.de>

### CODE REVIEW

**Comment by shibumi** [comment]
Hi this is my second PR. I hope this is enough for implementing another option for `systemd-resolved`.
@bastelfreak  can you have a look over it?

File: manifests/resolved.pp [hunk 1]
```diff
@@ -176,6 +180,23 @@
     }
   }
 
+  $_dnsovertls = $dnsovertls ? {
+    true    => 'opportunistic',
+    false   => 'no',
+    default => $dnsovertls,
+  }
+
+  if $_dnsovertls {
```
Comment (bastelfreak) [inline manifests/resolved.pp]: can you please have a look at the `spec/classes/init_spec.rb` file add add some spec tests for this change?

Comment (shibumi) [reply manifests/resolved.pp]: sure

**Comment by shibumi** [comment]
Tests should be implemented now ;)

**Pull request merged**